bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/internal/workbench/swt/OnboardingCTabFolder.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+* Copyright (c) 2022 SAP SE and others.
+*
+* This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License 2.0
+* which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-2.0/
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     SAP SE - initial version
+******************************************************************************/
+package org.eclipse.e4.ui.internal.workbench.swt;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.CTabFolder;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Listener;
+
+/**
+ * A CTabFolder which shows a set {@link Composite} in case there are no tabs.
```

Comment:
All what this (illegally) extended class does could be done by a legal SWT listener added to a regular CTabFolder. Please try this approach.

Reply:
Done